README.md

```diff
@@ -1,32 +1,505 @@
+  [Swrve sharedInstanceWithAppID:your_app_id apiKey:@"your_api_key" config:config launchOptions:launchOptions];
+  // Initialize Swrve with custom events that will trigger the push notification dialog
+  ```
+
+  If a user never triggers these events, they will never be available for push notifications. So select events that are most likely to be triggered by all but the most transient of users.
+
+### Processing custom payload
+
+If you want to capture any custom key/value pairs passed with the push notification, capture the notification data in both of your application’s `didFinishLaunchingWithOptions` and `didReceiveRemoteNotification` methods this:
+
+```
+- (BOOL) application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+   // Capture the notification data that opened the app
+   NSDictionary * remoteNotification = [launchOptions objectForKey:UIApplicationLaunchOptionsRemoteNotificationKey];
+   if (remoteNotification) {
+      // Do something with remoteNotification
+   }
+}
+
+-(void) application:(UIApplication*)application didReceiveRemoteNotification:(NSDictionary *)userInfo {
+    // Do something with userInfo
+}
+```
+
+The method above enables you specify any launch parameters that you want to capture and take action on. For more information, see [Configuring the Launch Screen](http://docs.swrve.com/faqs/push-notifications/configure-launch-screen-from-push-notification/).
+
+### Disabling method swizzling
+
+The SDK uses method swizzling by default. Method Swizzling of Push Notification methods means that fewer code changes are required to enable push notifications. Use the following code if you want to disable method swizzling:
+
+```
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+    SwrveConfig* config = [[SwrveConfig alloc] init];
+    config.pushEnabled = YES;
+    config.autoCollectDeviceToken = NO;
+    [[Swrve sharedInstance] initWithAppID:customerAppId apiKey:customerApiKey config:config launchOptions:launchOptions];
+}
+
+- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
+    if ([Swrve sharedInstance].talk != nil) {
+        [[Swrve sharedInstance].talk setDeviceToken:deviceToken];
+    }
+}
+
+-(void) application:(UIApplication*)application didReceiveRemoteNotification:(NSDictionary *)userInfo {
+    [[Swrve sharedInstance].talk pushNotificationReceived:userInfo];
+}
+```
+
+### Creating and uploading your iOS Certificate
+
+To enable your app to send push notifications to iOS devices, you require a push certificate. A push certificate authorizes an app to receive push notifications and authorizes a service to send push notifications to an app. For more information, see [How Do I Manage iOS Push Certificates?](http://docs.swrve.com/faqs/push-notifications/manage-ios-push-certificates-for-push-notifications/)
```

Comment:
A note on what happens if you change this cert? We get Zendesk tickets occasionally where push mysteriously stopped working for QA devices in a sandbox where they had updated their cert.